barretenberg/cpp/src/barretenberg/vm2/tracegen/tx_trace.cpp

```diff
@@ -243,100 +296,125 @@ void TxTraceBuilder::process(const simulation::EventEmitterInterface<simulation:
                                                              TransactionPhase::TEARDOWN,
                                                              TransactionPhase::COLLECT_GAS_FEES };
 
+    std::optional<simulation::TxStartupEvent> startup_event;
+
     for (const auto& tx_event : events) {
-        // Minus 1 since the enum is 1-indexed
-        phase_buckets[static_cast<uint8_t>(tx_event.phase) - 1].push_back(&tx_event);
+        if (std::holds_alternative<simulation::TxStartupEvent>(tx_event)) {
```

Comment:
A pity that this cannot be just another handled down below with the rest of them. Or can things be moved around so that it is?